onnx_xla/onnx_xla_interface.cc

```diff
@@ -221,49 +242,116 @@ ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
                     const onnxTensorDescriptor *inputDescriptors,
                     uint32_t outputsCount,
                     const onnxTensorDescriptor *outputDescriptors) {
-  try {
-    auto *executor = reinterpret_cast<onnx_xla::XlaExecutor*>(graph);
-    if (!executor) {
+  ONNXIFI_TRY_CATCH(([&, graph, inputsCount, inputDescriptors, outputsCount, outputDescriptors] {
+    if (!graph) {
       return ONNXIFI_STATUS_INVALID_GRAPH;
     }
     if (!inputDescriptors || !outputDescriptors) {
       return ONNXIFI_STATUS_INVALID_POINTER;
     }
-
+    auto *executor = reinterpret_cast<onnx_xla::XlaExecutor*>(graph);
     executor->initIO(inputsCount, inputDescriptors, outputsCount, outputDescriptors);
     return ONNXIFI_STATUS_SUCCESS;
-  }
-  ONNXIFI_CATCH_EXCPETION();
+  }))
 }
 
 //Runs the XlaExecutor by sending literals to server and executing computation 
 //TODO: support for synchronization primitives; For now assume, they are always set
 ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
     onnxRunGraph)(onnxGraph graph, const onnxMemoryFence *inputFence,
                   onnxMemoryFence *outputFence) {
-  try {
-    auto *executor = reinterpret_cast<onnx_xla::XlaExecutor *>(graph);
-    if (!executor) {
+  ONNXIFI_TRY_CATCH(([&, graph, inputFence, outputFence] {
+    if (!graph) {
       return ONNXIFI_STATUS_INVALID_GRAPH;
     }
-
+    auto *executor = reinterpret_cast<onnx_xla::XlaExecutor *>(graph);
     executor->sendLiterals();
     executor->executeComputation();
     return ONNXIFI_STATUS_SUCCESS;
-  }
-  ONNXIFI_CATCH_EXCPETION();
+  }))
 }
 
 //Frees executor memory
-ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
-ONNXIFI_SYMBOL_NAME(onnxReleaseGraph)(onnxGraph graph) {
-  try {
-    auto *executor = reinterpret_cast<onnx_xla::XlaExecutor *>(graph);
-    if (!executor) {
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxReleaseGraph)(onnxGraph graph) {
+  ONNXIFI_TRY_CATCH(([&, graph] {
+    if (!graph) {
       return ONNXIFI_STATUS_INVALID_GRAPH;
     }
+    auto *executor = reinterpret_cast<onnx_xla::XlaExecutor *>(graph);
     delete executor;
     return ONNXIFI_STATUS_SUCCESS;
-  }
-  ONNXIFI_CATCH_EXCPETION();
+  }))
+}
+
+//Initialize event by creating EventControl object on the heap
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI ONNXIFI_SYMBOL_NAME(
+    onnxInitEvent)(onnxBackend backend, onnxEvent* event) {
+  ONNXIFI_TRY_CATCH(([&, backend, event] {
```

Comment:
Yeah, here it's actually a bug? You are catching `event` by copy and and assign to an address to the newly created `EventControl` object. But this happens in the function closure. The the value of `event` of this function is not changed.

Reply:
fixed